src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -221,6 +222,84 @@
         ),
         terminator=OneOf("ORDER", "LIMIT", "HAVING", "QUALIFY", "WINDOW"),
     ),
+    PatternQuantifierGrammar=Sequence(
+        OneOf(
+            StringParser(
+                "+",
+                SymbolSegment,
+                name="plus",
+                type="plus",
+            ),
+            StringParser(
+                "*",
+                SymbolSegment,
+                name="star",
+                type="star",
+            ),
+            StringParser(
+                "?",
+                SymbolSegment,
+                name="question_mark",
+                type="question_mark",
```

Comment:
TIL I learned you could use these Parsers directly in Segments! Makes sense but never thought about doing that before.

Saying that, we have segments for each of these so why not do this as easier to read?:

```suggestion
            Ref("PositiveSegment"),
            Ref("StarSegment"),
            Ref("QuestionMarkSegment"),
```

Note the last one will need to be added (you can copy from BigQuery). Also `PositiveSegment` has a different type so if that matters you could add a new one.

Reply:
Think I was a bit concerned that the rules maybe moving things about since patterns don't follow the same formatting as expressions, let me have a play

Reply:
Sorry don't understand. The `Ref`s are exactly the same as the code that's currently in there?